File: quasar-backend/src/controllers/personaController.js
const Persona = require('../models/Persona');
const logger = require('../utils/logger');

/**
 * GET /api/personas
 * Returns all available interview personas
 */
async function getPersonas(req, res) {
  try {
    const personas = await Persona.find()
      .select('_id name description interruptionStyle followUpAggression')
      .sort({ followUpAggression: 1 })
      .lean();

    // Map to match frontend expectations (snake_case)
    const result = personas.map(p => ({
      id: p._id,
      name: p.name,
      description: p.description,
      interruption_style: p.interruptionStyle,
      follow_up_aggression: p.followUpAggression,
    }));

    return res.json({
      success: true,
      message: 'Personas retrieved',
      data: result
    });
  } catch (err) {
    logger.error('Get personas error', { err: err.message });
    return res.status(500).json({ success: false, message: 'Failed to get personas', data: null });
  }
}

/**
 * GET /api/personas/:id
 * Returns a single persona with full system prompt
 */
async function getPersona(req, res) {
  try {
    const { id } = req.params;

    const persona = await Persona.findById(id).lean();

    if (!persona) {
      return res.status(404).json({ success: false, message: 'Persona not found', data: null });
    }

    return res.json({
      success: true,
      message: 'Persona retrieved',
      data: {
        id: persona._id,
        name: persona.name,
        description: persona.description,
        system_prompt: persona.systemPrompt,
        interruption_style: persona.interruptionStyle,
        follow_up_aggression: persona.followUpAggression,
      }
    });
  } catch (err) {
    logger.error('Get persona error', { err: err.message });
    return res.status(500).json({ success: false, message: 'Failed to get persona', data: null });
  }
}

module.exports = { getPersonas, getPersona };

File: backend/src/main/java/com/teamzemo/quasar/controller/PersonaController.java
package com.teamzemo.quasar.controller;

import com.teamzemo.quasar.dto.ApiResponse;
import com.teamzemo.quasar.model.Persona;
import com.teamzemo.quasar.repository.PersonaRepository;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.LinkedHashMap;
import java.util.List;
import java.util.Map;
import java.util.stream.Collectors;

@RestController
@RequestMapping("/api/personas")
public class PersonaController {

    private static final Logger log = LoggerFactory.getLogger(PersonaController.class);

    private final PersonaRepository personaRepository;

    public PersonaController(PersonaRepository personaRepository) {
        this.personaRepository = personaRepository;
    }

    /** GET /api/personas */
    @GetMapping
    public ResponseEntity<ApiResponse<List<Map<String, Object>>>> getPersonas() {
        try {
            List<Map<String, Object>> result = personaRepository.findAllByOrderByFollowUpAggressionAsc()
                    .stream()
                    .map(p -> {
                        Map<String, Object> m = new LinkedHashMap<>();
                        m.put("id",                  p.getId());
                        m.put("name",                p.getName());
                        m.put("description",         p.getDescription());
                        m.put("interruption_style",  p.getInterruptionStyle());
                        m.put("follow_up_aggression", p.getFollowUpAggression());
                        return m;
                    })
                    .collect(Collectors.toList());
            return ResponseEntity.ok(ApiResponse.ok("Personas retrieved", result));
        } catch (Exception e) {
            log.error("Get personas error", e);
            return ResponseEntity.internalServerError().body(ApiResponse.fail("Failed to get personas"));
        }
    }

    /** GET /api/personas/:id */
    @GetMapping("/{id}")
    public ResponseEntity<ApiResponse<Map<String, Object>>> getPersona(@PathVariable String id) {
        try {
            Persona persona = personaRepository.findById(id).orElse(null);
            if (persona == null) {
                return ResponseEntity.status(404).body(ApiResponse.fail("Persona not found"));
            }
            Map<String, Object> data = new LinkedHashMap<>();
            data.put("id",                  persona.getId());
            data.put("name",                persona.getName());
            data.put("description",         persona.getDescription());
            data.put("system_prompt",       persona.getSystemPrompt());
            data.put("interruption_style",  persona.getInterruptionStyle());
            data.put("follow_up_aggression", persona.getFollowUpAggression());
            return ResponseEntity.ok(ApiResponse.ok("Persona retrieved", data));
        } catch (Exception e) {
            log.error("Get persona error", e);
            return ResponseEntity.internalServerError().body(ApiResponse.fail("Failed to get persona"));
        }
    }
}
